src/Chainweb/Mempool/InMem.hs

```diff
@@ -488,8 +488,8 @@ getBlockInMem logg cfg lock (BlockFill gasLimit txHashes _)  txValidate bheight
         -- put the txs chosen for the block back into the map -- they don't get
         -- expunged until they are mined and validated by consensus.
         let !psq'' = V.foldl' ins (HashMap.union seen psq') out
-        writeIORef (_inmemPending mdata) $ psq''
-        writeIORef (_inmemBadMap mdata) $ badmap'
+        writeIORef (_inmemPending mdata) $! force psq''
```

Comment:
It's `$! force` redundant? `force` is already evaluating to HNF, so why do we then need to evaluate that to WHNF?

Reply:
Yes, it is. Choose `$ force psq''` or `$!! psq''` but not both!

Reply:
Are we sure? `force` when evaluated evaluates it's argument to nf. Does `$ force` necessarily trigger the evaluation of `force` itself? I guess, `force` include `seq` at the top-level which gets special treatment by GHC. But I haven't checked the code.